circuit-codegen/src/main/kotlin/com/slack/circuit/codegen/CircuitSymbolProcessorProvider.kt

```diff
@@ -132,44 +162,66 @@ private class CircuitSymbolProcessor(
           } else {
             FactoryType.UI
           }
```

Comment:
I think it's not going to be a big issue, especially as we have #149 too that could be used to help guide this. Let's solve it if it becomes an issue later rather than have multiple annotations now, as I think the overhead of different types exceeds the cost of remembering to write "Presenter" in the name currently. We can also put more guard rails in code gen itself to make sure things are structured right (such as requiring presenter functions to return states, etc)

Reply:
Ok, sounds good 👍